scipy/linalg/basic.py

```diff
@@ -1132,54 +1132,81 @@ def balance(A, permute=True, scale=True, separate = False, overwrite_a=False):
     a similarity transformation such that the magnitude variation of the 
-    matrix entries is reflected to the scaling matrices. This is particularly 
-    useful for eigenvalue and matrix decompositions and often automatically 
-    performed. Hence, this function provides a convenience for manually 
-    balancing. 
+    matrix entries is reflected to the scaling matrices. 
         
-    Moreover, if turned on, the matrix is first permuted to isolate the upper 
-    triangular parts of the matrix and only the remaining part is scaled. 
+    Moreover, if enabled, the matrix is first permuted to isolate the upper 
+    triangular parts of the matrix and, again if scaling is also enabled,
+    only the remaining subblocks are subjected to scaling. 
     
     The balanced matrix satisfies the following equality 
     
     .. math::
          
                         B = T^{-1} A T 
 
-    The algorithm is based on the well-known technique of [1]_ and has 
-    been modified to account for special cases. See [2]_ for details
-    which have been implemented since LAPACK v3.5.0. 
+    The scaling coefficients are approximated to the nearest power of 2 
+    to avoid round-off errors.
     
     Parameters
     ----------
     A : (n, n) array_like
         Square data matrix for the balancing.
-    permute : bool
+    permute : bool, optional
         The selector to define whether permutation of A is also performed
         prior to scaling.
-    scale : bool
+    scale : bool, optional
         The selector to turn on and off the scaling. If False, the matrix 
         will not be scaled. 
-    separate : bool
+    separate : bool, optional
         This switches from returning a full matrix of the transformation
         to a tuple of two separate 1D permutation and scaling arrays.
-    overwrite_a : bool
+    overwrite_a : bool, optional
         This is passed to xGEBAL directly. Essentially, overwrites the result
         to the data. It might increase the space efficiency. See LAPACK manual 
         for details. This is False by default. 
 
     Returns
     -------
-    B : (n, n) array
+    B : (n, n) ndarray
         Balanced matrix 
-    T : (n, n) array
+    T : (n, n) ndarray
         A possibly permuted diagonal matrix whose nonzero entries are 
         integer powers of 2 to avoid numerical truncation errors. 
-    s, p : (n,) array
-        If ``separate`` keyword is set to True then instead of the matrix 
-        ``T`` above, the scaling and the permutation vector is given. 
+    scale, perm : (n,) ndarray
+        If ``separate`` keyword is set to True then instead of the array 
+        ``T`` above, the scaling and the permutation vector is given 
+        separately without allocating the full array ``T``. 
 
     .. versionadded:: 0.19.0
+
+    Notes
+    -----
+
+    This algorithm is particularly useful for eigenvalue and matrix 
+    decompositions and often automatically performed internally. 
```

Comment:
often in the sense that "in many lapack routines this is the case" but now come to think of it I sense it is a bit off indeed.